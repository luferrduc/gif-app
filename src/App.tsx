

export const App = () => {
  return (
    <>
      <h1>Gif Expert App</h1>
    
    </>
  )
}