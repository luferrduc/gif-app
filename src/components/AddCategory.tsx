import { ChangeEvent, FormEvent, useState } from 'react';
import './AddCategory.css'

export const AddCategory = () => {
  const [inputValue, setInputValue] = useState('One Punch');

  const onInputChange = ({ target }: ChangeEvent<HTMLInputElement>) => {
    setInputValue(target.value)
  }

  const onSubmit = (event: FormEvent) => {
    event.preventDefault()
  }

  return (
    <form className="form" onSubmit={onSubmit}>
      <input 
        type="text" 
        id="input-category"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={ onInputChange }
      />
    </form>
  )
}