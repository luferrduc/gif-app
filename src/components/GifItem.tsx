interface Props {
  id?: string;
  title: string;
  url: string;
}

export const GifItem = ({ title, url }: Props): JSX.Element => {
  return (
    <div className="card">
      <p>{title}</p>
      <img src={url} alt={title} />
    </div>
  )
}