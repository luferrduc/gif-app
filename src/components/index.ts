// export { AddCategory } from "./AddCategory";
// export { GifGrid } from "./GifGrid";
// export { GifItem } from "./GifItem";
// export { Loading } from "./Loading";

export * from "./AddCategory";
export * from "./GifGrid";
export * from "./GifItem";
export * from "./Loading";

