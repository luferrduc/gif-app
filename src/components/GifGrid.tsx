import { GifItem } from "./GifItem"
import { useFetchGifs } from "@/hooks/useFetchGifs"
import { Loading } from "./Loading"

interface Props {
	category: string
}

export const GifGrid = ({ category }: Props) => {
	const { images, isLoading } = useFetchGifs(category)

	
	return (
		<section className="gif-section">
			<h3 className="title">{category}</h3>
			<article className="card-grid">
				
				{ isLoading && <Loading />}
				{ images.map((image) => (<GifItem key={image.id} {...image} />)) }

			</article>
		</section>
	)
}

