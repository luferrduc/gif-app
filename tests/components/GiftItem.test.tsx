import React from 'react'
import { render } from "@testing-library/react"
import { GifItem } from "../../src/components"
// import { GifItem } from '@co'

describe('Se prueba <GifItem />', () => { 
  
  const images = {
    id: "",
    title: "Dragon Ball Gif",
    url: "https://giphy.com/gifs/dragon_ball.gif"
  }

  test('debe hacer match con el snapshot', () => { 
      const { title, url } = images
      const { container } = render(<GifItem key={images.id} title={title} url={url} />)
      expect(container).toMatchSnapshot()
  })
})