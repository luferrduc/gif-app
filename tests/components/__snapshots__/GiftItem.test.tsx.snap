// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Se prueba <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Dragon Ball Gif
    </p>
    <img
      alt="Dragon Ball Gif"
      src="https://giphy.com/gifs/dragon_ball.gif"
    />
  </div>
</div>
`;
